mlflow/xgboost.py

```diff
@@ -434,26 +434,75 @@ def log_feature_importance_plot(features, importance, importance_type):
             Log feature importance plot.
             """
             import matplotlib.pyplot as plt
+            from cycler import cycler
 
             features = np.array(features)
-            importance = np.array(importance)
-            indices = np.argsort(importance)
-            features = features[indices]
-            importance = importance[indices]
+
+            # Structure the supplied `importance` values as a `num_features`-by-`num_classes` matrix
+            importances_per_class_by_feature = np.array(importance)
+            if importances_per_class_by_feature.ndim <= 1:
+                # In this case, the supplied `importance` values are not given per class. Rather,
+                # one importance value is given per feature. For consistency with the assumed
+                # `num_features`-by-`num_classes` matrix structure, we coerce the importance
+                # values to a `num_features`-by-1 matrix
+                indices = np.argsort(importance)
+                features = features[indices]
+                importances_per_class_by_feature = np.array(
+                    [[importance] for importance in importances_per_class_by_feature]
+                )
```

Comment:
```suggestion
                importances_per_class_by_feature = np.array(
                    [[importance] for importance in importances_per_class_by_feature[indices]]
                )
```

Can we sort `importance` as well?

Reply:
Thanks for catching this! Done! Here's a screenshot from tree booster on MNIST:

![feature_importance_weight](https://user-images.githubusercontent.com/39497902/124563297-c44be900-ddf4-11eb-95e9-d17986edfc1c.png)